### PR TITLE
Implement array to array functionality to `chainerx.maximum`

### PR DESCRIPTION
**Pull request by aksub99**
Implementation of array vs array functionality to chainerx.maximum
Ref. #6423 
TODO: Add test for python binding

### CODE REVIEW

File: chainerx_cc/chainerx/routines/math.cc [hunk 1]
```diff
@@ -531,6 +562,10 @@ Array Maximum(const Array& x1, Scalar x2) {
 
 Array Maximum(Scalar x1, const Array& x2) { return Maximum(x2, x1); }
 
+Array Maximum(const Array& x1, const Array& x2) {
+    return IfLessElse(x1, x2, x2, x1);  // x1 < x2 ? x2 : x1
```
Comment (asi1024) [inline chainerx_cc/chainerx/routines/math.cc]: How about writing as `IfGreaterElse(x1, x2, x1, x2)` if you will complete #6541 first?

Comment (asi1024) [reply chainerx_cc/chainerx/routines/math.cc]: Merged #6541.

**Comment by aksub99** [comment]
Sorry, made the above merge commits by mistake. Will undo.

**Comment by aksub99** [comment]
@asi1024 It seems as though that the travis failures are due to tight tolerances. Could they be relaxed?

File: chainerx_cc/chainerx/cuda/cuda_device.h [hunk 2]
```diff
@@ -109,6 +109,7 @@ class CudaDevice : public Device {
     // activation.cu
 
     void IfLessElseASSA(const Array& x1, Scalar x2, Scalar pos, const Array& neg, const Array& out) override;
+    void IfLessElseAAAA(const Array& x1, const Array& x2, const Array& pos, const Array& neg, const Array& out) override;
```
Comment (asi1024) [inline chainerx_cc/chainerx/cuda/cuda_device.h]: This member function seems unused. Could you remove this?

Comment (aksub99) [reply chainerx_cc/chainerx/cuda/cuda_device.h]: Sure! Working on it!

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 3]
```diff
@@ -2034,6 +2034,45 @@ def test_max_invalid_shapes_and_axis(device, array, axis, dtype, is_module):
...
+        'in_dtypes,out_dtype': (
+            _make_same_in_out_dtypes(2, chainerx.testing.numeric_dtypes)),
+        'input_lhs': ['random'],
+        'input_rhs': ['random'],
+        'is_module': [True, False],
+    })
+    # TODO(aksub99): Add tests for inf and NaN.
+))
+class TestMaximum(BinaryMathTestBase, op_utils.NumpyOpTest):
+
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Could you add this line? This function has non-differentiable points.
```suggestion

    dodge_nondifferentiable = True

```

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 4]
```diff
@@ -2034,6 +2034,45 @@ def test_max_invalid_shapes_and_axis(device, array, axis, dtype, is_module):
             a.max(axis)
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'in_dtypes,out_dtype': (
+            _make_same_in_out_dtypes(2, chainerx.testing.numeric_dtypes)),
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Could you add mixed-dtypes tests? (Sorry, I should have said when reviewing `TestMinimum`.)

**Comment by asi1024** [comment]
CI failure is unrelated to this PR. Fixed it in #6858.

**Comment by asi1024** [comment]
Could you resolve conflicts?

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM! Thank you very much!

**Pull request merged**